### PR TITLE
Coreloader

### PR DESCRIPTION
**Pull request by gimhael**
This changes GLEW to work with core and forward compatible profiles and also skip querying function pointers that have been removed in those profiles.

Note: I haven't compiled or tested the wgl part, I just made the same changes as for glx.

### CODE REVIEW

File: auto/src/glew.gperf [hunk 1]
```diff
@@ -0,0 +1,674 @@
+%struct-type
```
Comment (nigels-com) [inline auto/src/glew.gperf]: Generally speaking, generated files do not typically get committed to the repository.

File: auto/src/glewinfo_tail.c [hunk 2]
```diff
@@ -11,35 +11,44 @@ int main (void)
...
-  if (GL_TRUE == glewCreateContext(display, &visual))
-#endif
+  if (GL_TRUE == glewCreateContext(&params))
   {
     fprintf(stderr, "Error: glewCreateContext failed\n");
     glewDestroyContext();
     return 1;
   }
-  glewExperimental = GL_TRUE;
+  //glewExperimental = GL_TRUE;
```
Comment (nigels-com) [inline auto/src/glewinfo_tail.c]: If there is a reason for this to change, a comment for explanation, please.

**Comment by nigels-com** [comment]
If I'm interpreting this correctly - this is for core profiles, but no longer supports compatibility?

File: auto/src/glew.gperf (hunk 1 above)
Comment (gimhael) [inline auto/src/glew.gperf]: Ok, I will remove it.

File: auto/src/glewinfo_tail.c (hunk 2 above)
Comment (gimhael) [inline auto/src/glewinfo_tail.c]: I changed that for testing and shouldn't have commited it, I will revert the change.

**Comment by gimhael** [comment]
The change always uses glGetStringi for GL versions >= 3.0, which is available on compatibility profiles too, so it works on both core and compatibility profiles. It falls back to the old glGetString method on older profiles.

**Comment by nigels-com** [comment]
I did some work with the glewinfo part of this to make it work on Mac OS X.
Would you mind taking a look at my pull request, and try it on Linux?
Note that I refined the command-line to accept slightly different (better?) parameters.

https://github.com/nigels-com/glew/pull/43

**Comment by nigels-com** [comment]
Oh, and if you don't mind rebasing onto that too... :-)

**Comment by gimhael** [comment]
I rebased onto the glewinfo-core branch, and it works on Linux (both the pure glewinfo-core branch and the rebased PR).

**Comment by nigels-com** [comment]
Great!  Thanks.

**Comment by nigels-com** [comment]
I've rebased this branch onto the current master branch, which incorporates some core context support.  There is more here that I'd like to bring to master branch.  I'm not sure my rebase was 100% accurate, due to all the churn on master.  However, it's still a useful reference for what GLEW ought to be doing in terms of core profile support.

**Comment by nigels-com** [comment]
https://github.com/nigels-com/glew/tree/gimhael

**Comment by nigels-com** [comment]
I'm closing this pull request since it mostly overlaps with what was done for GLEW 2.0.0.  This contribution was a valuable contribution to core context support, even if it wasn't merged as-is.  Thank you.

**Pull request closed without merging**